pkg/services/pluginsintegration/pluginconfig/envvars.go

```diff
@@ -174,6 +177,9 @@ func (p *EnvVarsProvider) pluginSettingsEnvVars(pluginID string) []string {
 	return env
 }
 
-func envVar(key, value string) string {
+func (p *EnvVarsProvider) envVar(key, value string) string {
+	if strings.Contains(value, "\x00") {
+		p.logger.Error("Variable with key '%s' contains a nil character", key)
```

Comment:
I am missing a bit of error handling here. It would help to know what's the error you are seeing. Is it a mistake to set variables with the value `\x00` or is it a problem in the reader of the variable?

If it's not an error, I would set this log level to `Debug` (which still let's you debug the issue if it appears). If it's an error, I would return the error here rather than continuing with the current behavior.

Also, a unit test with what you expect would be nice.